controllers/components/modelmeshserving/modelmesh_support.go

```diff
@@ -0,0 +1,87 @@
+package modelmeshserving
+
+import (
+	"strings"
+
+	appsv1 "k8s.io/api/apps/v1"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+)
+
+// a workaround for 2.5 due to odh-model-controller serivceaccount keeps updates with label.
+var saPredicates = predicate.Funcs{
+	UpdateFunc: func(e event.UpdateEvent) bool {
+		namespace := e.ObjectNew.GetNamespace()
+		if e.ObjectNew.GetName() == DependentComponentName && (namespace == "redhat-ods-applications" || namespace == "opendatahub") {
+			return false
+		}
+		return true
+	},
+}
+
+// a workaround for modelmesh and kserve both create same odh-model-controller NWP.
```

Comment:
I guess these predicates are for cases where both modelmesh and kserve are installed at the same time, and I should be adding similar predicates to resources on the kserve side?

Reply:
i was hoping all these can be moved into dependency, so no need duplicated in both kserve and modelmesh.

Reply:
By dependency, do you mean like having odh-model-controller having its own API and controller? I guess that would be up to the DSC controller to check if at least one of kserve and/or modelmesh are managed, and then install and manage the model controller?

Reply:
this one https://issues.redhat.com/browse/RHOAIENG-15481